playgrounds/vite-react/src/App.tsx

```diff
@@ -255,8 +259,15 @@ function SendTransaction() {
     sendTransaction({ to, value: parseEther(value) })
   }
 
-  const { isLoading: isConfirming, isSuccess: isConfirmed } =
+  const { isLoading: isConfirming, data: receipt } =
     useWaitForTransactionReceipt({
+      config: {
+        ...config,
+        // Additionally, we can pass a client to the config
+        // to use a different client than the one passed to the hook
+        client,
+      },
+      chainId: connectorClientData?.chain?.id,
```

Comment:
assuming this will help someone in the future when they run into / search this issue.